pkg/apis/core/validation/shoot.go

```diff
@@ -717,9 +717,20 @@ func validateNodeLocalDNSUpdate(newSpec, oldSpec *core.ShootSpec, fldPath *field
 	)
 
 	if oldNodeLocalDNSEnabled != newNodeLocalDNSEnabled {
+		defaultVersion, err := semver.NewVersion(oldSpec.Kubernetes.Version)
+		if err != nil {
+			return field.ErrorList{field.Invalid(fldPath, oldSpec.Kubernetes.Version, fmt.Sprintf("failed to parse old Kubernetes version %q: %v", oldSpec.Kubernetes.Version, err))}
```

Comment:
`fldPath` is not correct. It points to `spec`, but should point to `spec.kubernetes.version`.

```suggestion
			return field.ErrorList{field.Invalid(fldPath.Child("kubernetes", "version"), oldSpec.Kubernetes.Version, fmt.Sprintf("failed to parse old Kubernetes version %q: %v", oldSpec.Kubernetes.Version, err))}
```